pkg/apis/componentconfig/validation/validation.go

```diff
@@ -68,8 +68,12 @@ func ValidateSchedulerAlgorithmSource(cc *componentconfig.SchedulerAlgorithmSour
 // ValidateSchedulerPolicySource ensures validation of the SchedulerPolicySource struct
 func ValidateSchedulerPolicySource(cc *componentconfig.SchedulerPolicySource, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if cc.File == nil && cc.ConfigMap == nil {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("file"), nil, "needs to set either File or ConfigMap"))
+	if cc == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("policy"), nil, "need to set Policy"))
```

Comment:
could be refactored as:

```
allErrs := field.ErrorList{}
if cc == nil {
	return append(allErrs, field.Invalid(fldPath.Child("policy"), nil, "need to set Policy"))
}
if cc.File == nil && cc.ConfigMap == nil {
	return append(allErrs, field.Invalid(fldPath.Child("file"), nil, "need to set either File or ConfigMap"))
}
return allErrs
```
to avoid indenting the `} else` part.